### PR TITLE
core: add package level javadoc.

### LINKED ISSUE #714: Core gRPC package needs documention
The default javadoc generated is missing package level documentation

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
Fixes #714

### CODE REVIEW

File: core/src/main/java/io/grpc/package-info.java [hunk 1]
```diff
@@ -29,7 +29,43 @@
...
  */
 
+
 /**
  * The gRPC core public API.
+ *
+ * <p>gRPC is based on a client-server model of remote procedural calls.  A client creates a
+ * channel which is connected to a server.  RPCs are initiated from the client and sent to the
+ * server which then responds back to the client.  When the client and server are done sending
+ * messages, they half close their respective connections.  When both sides are done, the RPC is
```
Comment (ejona86) [inline core/src/main/java/io/grpc/package-info.java]: The RPC is complete as soon as the server closes.

File: core/src/main/java/io/grpc/package-info.java [hunk 2]
```diff
@@ -29,7 +29,43 @@
...
+ * complete.
+ *
+ * <p>To send an RPC, first create a {@link Channel} using {@link ManagedChannelBuilder#forTarget}.
+ * When using auto generate Protobuf stubs, the stub class will have constructors for wrapping the
```
Comment (ejona86) [inline core/src/main/java/io/grpc/package-info.java]: s/auto generate/auto-generated/

**Comment by ejona86** [comment]
@carl-mastrangelo LGTM

File: core/src/main/java/io/grpc/package-info.java (hunk 1 above)
Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/package-info.java]: Done.

File: core/src/main/java/io/grpc/package-info.java (hunk 2 above)
Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/package-info.java]: Done.

**Comment by carl-mastrangelo** [comment]
This is very silly, but package level javadoc references don't resolve (and get a build time warning) if the "package io.grpc" line comes after the javadoc.  I am moving it up rather than fully resolving the references.

**Pull request merged**